components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -151,20 +218,21 @@ int GetNormal(int min, int max) {
   return min + floor((max - min) * GetNormal());
 }
 
-using ShouldConsiderPredicate = bool(const Signal& signal);
+using GetWeighting = double(const ArticleWeight& weight);
 
 // Picks an article with a probability article_weight/sum(article_weights).
 mojom::FeedItemMetadataPtr PickRouletteAndRemove(
     ArticleInfos& articles,
-    ShouldConsiderPredicate should_consider = [](const auto& signal) {
-      return true;
+    GetWeighting get_weighting = [](const auto& weight) {
+      return weight.weighting;
     }) {
   double total_weight = 0;
-  for (const auto& [article, signal, weight] : articles) {
-    if (!should_consider(signal)) {
+  for (const auto& [article, weight] : articles) {
+    auto weighting = get_weighting(weight);
+    if (weighting == 0) {
       continue;
     }
-    total_weight += weight;
+    total_weight += weighting;
```

Comment:
I think simply adding the `weight.weighting` without checking would be okay

```suggestion
    total_weight += weight.weiting;
```

Reply:
:facepalm: I think I had another branch in there for disabled sources at one point and didn't remove the check :laughing: Nice catch